crates/sui-indexer-alt-jsonrpc/src/lib.rs

```diff
@@ -250,6 +253,15 @@ pub async fn start_rpc(
     rpc.add_module(QueryTransactions(context.clone(), transactions_config))?;
     rpc.add_module(Transactions(context.clone()))?;
 
+    // Add the write module if a fullnode rpc url is provided.
+    if let Some(write_args) = write_args {
+        let default_write_config = WriteConfig::default();
+        let write_config = write.map_or(default_write_config.clone(), |w| {
+            w.finish(default_write_config)
+        });
```

Comment:
With the suggestion to not make `write` optional, you can simplify this code. If you could also then move it up to where the other `.finish` calls are, which will help deal with the merge conflict with #21332.